src/Instagram.php

```diff
@@ -76,9 +76,19 @@ public function refreshToken($token)
         return $this->http->get($url);
     }
 
-    public function fetchMedia(AccessToken $token, $limit = 20)
+    public function fetchMedia(AccessToken $token, $limit = null)
     {
-        $url = sprintf(self::MEDIA_URL_FORMAT, $token->user_id, self::MEDIA_FIELDS, $limit, $token->access_code);
+        $queryLimit = $limit; //optimal number of requests
+
+        if ($limit === null || $limit > 1000) {
+            $queryLimit = 100;
+        } elseif ($limit <= 100) {
+            $queryLimit = $limit;
+        } elseif ($limit > 100) {
+            $queryLimit = (int) ceil( $queryLimit/ ceil($queryLimit/100) );
```

Comment:
calculates optimal $queryLimit unless over 1000 in which case will be 100 per request